src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2,13 +2,160 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
-
+using System.Numerics;
+using System.Runtime.CompilerServices;
 using Internal.Runtime.CompilerServices;
 
 namespace System
 {
     internal static partial class SpanHelpers // .T
     {
+        public static void Fill<T>(ref T refData, uint numElements, T value)
+        {
+            // n.b. If Fill is ever adapted to work for lengths beyond uint.MaxValue, double-check
+            // where it's used in the arithmetic operations below to ensure no integer overflow is possible.
+
+            if (numElements == 0)
+            {
+                return; // nothing to do
+            }
+
+            if (typeof(T) != typeof(float) && typeof(T) != typeof(double) && !RuntimeHelpers.IsBitwiseEquatable<T>())
+            {
+                goto CannotVectorize; // it might not be valid to SIMD-spray this value into the backing span
+            }
+
+            if (Vector.IsHardwareAccelerated)
+            {
+                if (numElements < (uint)(Vector<byte>.Count / Unsafe.SizeOf<T>()))
+                {
+                    goto CannotVectorize; // not enough data for even a single run through the vectorized loop
+                }
+
+                Vector<byte> vector;
+
+                if (typeof(T) == typeof(float))
+                {
+                    vector = (Vector<byte>)(new Vector<float>((float)(object)value!));
+                }
+                else if (typeof(T) == typeof(double))
+                {
+                    vector = (Vector<byte>)(new Vector<double>((double)(object)value!));
+                }
+                else
+                {
+                    T tmp = value; // Avoid taking address of the "value" argument. It would regress performance of the loops below.
+                    if (Unsafe.SizeOf<T>() == 1)
+                    {
+                        vector = new Vector<byte>(Unsafe.As<T, byte>(ref tmp));
+                    }
+                    else if (Unsafe.SizeOf<T>() == 2)
+                    {
+                        vector = (Vector<byte>)(new Vector<ushort>(Unsafe.As<T, ushort>(ref tmp)));
+                    }
+                    else if (Unsafe.SizeOf<T>() == 4)
+                    {
+                        vector = (Vector<byte>)(new Vector<uint>(Unsafe.As<T, uint>(ref tmp)));
+                    }
+                    else if (Unsafe.SizeOf<T>() == 8)
+                    {
+                        vector = (Vector<byte>)(new Vector<ulong>(Unsafe.As<T, ulong>(ref tmp)));
+                    }
+                    else
```

Comment:
This seems like it should handle `Unsafe.SizeOf<T>() == 16`, it could at least handle `Vector2`, `Vector4`, `Vector128<T>`, `Guid`, and `decimal`, but potentially other types as well.

Reply:
How do I get an arbitrary 16-byte T into a `Vector<T>`? There doesn't seem to be any relevant ctor or cast I can perform.

Reply:
It would basically be:
```csharp
var v128 = Unsafe.As<T, Vector128<byte>>(ref value);

if (sizeof(Vector<T>) == 32)
{
    var v256 = Vector256.Create(v128, v128);
    vector = v256.AsVector();
}
else
{
    Debug.Assert(sizeof(Vector<T>) == 16);
    vector = v128.AsVector();
}
```